pkg/sql/schemachanger/scbuild/internal/scbuildstmt/helpers.go

```diff
@@ -776,12 +776,12 @@ func makeSwapIndexSpec(
 	return in, temp
 }
 
-// fallBackIfZoneConfigExists determines if the table has regional by row
-// properties and throws an unimplemented error.
-func fallBackIfZoneConfigExists(b BuildCtx, n tree.NodeFormatter, id catid.DescID) {
+// fallBackIfSubZoneConfigExists determines if the table has a subzone
+// config.
+func fallBackIfSubZoneConfigExists(b BuildCtx, n tree.NodeFormatter, id catid.DescID) {
 	{
 		tableElts := b.QueryByID(id)
-		if _, _, elem := scpb.FindTableZoneConfig(tableElts); elem != nil {
+		if _, _, elem := scpb.FindIndexZoneConfig(tableElts); elem != nil {
 			panic(scerrors.NotImplementedErrorf(n,
 				"regional by row partitioning is not supported"))
```

Comment:
This error message need to changed too :) Also some commentary can be used to explain why sub-zones matter.

Reply:
Good point added an explanation here too:
// Normally this logic is used to limit index-related operations,
// since dropping indexes will need to remove entries of sub zones from
// the zone config.